src/actions/RequestReviewActions.js

```diff
@@ -0,0 +1,66 @@
+// @flow
+
+import React from 'react'
+import { Linking, Platform } from 'react-native'
+import * as StoreReview from 'react-native-store-review'
+import { sprintf } from 'sprintf-js'
+
+import { TwoButtonSimpleConfirmationModal } from '../components/modals/TwoButtonSimpleConfirmationModal'
+import { Airship } from '../components/services/AirshipInstance'
+import s from '../locales/strings'
+import * as CORE_SELECTORS from '../modules/Core/selectors'
+import type { State } from '../types/reduxTypes.js'
+
+const SWAP_COUNT_DATA_FILE = 'swapCountData.json'
+const MANY_SWAPS_TO_TRIGGER_REQUEST = 3
+
+const requestReview = async () => {
+  if (Platform.OS === 'ios') {
+    StoreReview.requestReview()
+  } else if (Platform.OS === 'android') {
+    const title = sprintf(s.strings.request_review_question_title, s.strings.app_name_short)
+    await Airship.show(bridge => (
+      <TwoButtonSimpleConfirmationModal
+        bridge={bridge}
+        title={title}
+        subTitle={s.strings.request_review_question_subtitle}
+        cancelText={s.strings.request_review_answer_no}
+        doneText={s.strings.request_review_answer_yes}
+        onCancel={() => {}}
+        onDone={() => Linking.openURL(s.strings.request_review_android_page_link)}
```

Comment:
This modal should not have `onCancel` or `onDone` callbacks. Instead, it should use the promise return value to indicate which button the user pressed.

Reply:
@swansontec Can you please provide an example in the edge-react-ui code of what you're talking about?  I copied this pattern from the other modals in that same directory.  I don't see any other modals that return boolean-resolved-promises.  I'd like to be imitating your preexisting style.  All the other modals appear to be using callbacks.